src/ipc/ipc4/helper.c

```diff
@@ -139,8 +139,8 @@ __cold struct comp_dev *comp_new_ipc4(struct ipc4_module_init_instance *module_i
 	ipc_config.ipc_config_size = module_init->extension.r.param_block_size * sizeof(uint32_t);
 	ipc_config.ipc_extended_init = module_init->extension.r.extended_init;
 
-	dcache_invalidate_region((__sparse_force void __sparse_cache *)MAILBOX_HOSTBOX_BASE,
-				 MAILBOX_HOSTBOX_SIZE);
+	sys_cache_data_invd_range((__sparse_force void __sparse_cache *)MAILBOX_HOSTBOX_BASE,
+				 ipc_config.ipc_config_size);
```

Comment:
These are good additions. This is overlapping a bit with my IPC payload access rework https://github.com/thesofproject/sof/pull/10346 but that's stuck now due to dependencies, so no problem with you modifying these here.